File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const obj = {};
        let diff;
        let res = [];
        nums.forEach((item, i) => {
            diff = target - item;
            if (obj[diff] >=0 ) {
                res.push(i, obj[diff]);
            }
            else {
                obj[item] = i;
            }
        })

        return res.sort();
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)):
            product = 1
            for j in nums[:i]:
                product = product * j
            for j in nums[i+1:]:
                product = product * j
            res.append(product)
        return res
